server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -901,6 +932,16 @@ protected void abandonSegment(final long truncatedTime, final Sink sink)
           sink.getVersion(),
           new SingleElementPartitionChunk<>(sink)
       );
+      Iterables.transform(sink, new Function<FireHydrant, Void>()
+      {
+        @Nullable
+        @Override
+        public Void apply(@Nullable FireHydrant input)
+        {
+          cache.close(makeHydrantIdentifier(input, input.getSegment()));
+          return null;
+        }
+      });
```

Comment:
`Iterables.transform` is lazy so I think this code does not actually do anything.